components/typeahead/typeahead.directive.ts

```diff
@@ -58,82 +58,96 @@ export class Typeahead implements OnInit {
               private loader:DynamicComponentLoader) {
   }
 
-  public get matches() {
-    return this._matches;
-  }
+  private asyncActions() {
+    let s = new Subject();
+    s.subscribe(() => {
+      this.typeahead
+        .subscribe(
+          (matches:string[]) => {
+            this._matches = matches.slice(0, this.typeaheadOptionsLimit);
```

Comment:
@buchslava `matches.slice(0, this.typeaheadOptionsLimit)` -> `.take(this.typeaheadOptionsLimit)`